lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -57,7 +57,10 @@ def include_meta(json)
 
         def resource_object_for(options)
           cache_check(serializer) do
-            serializer.attributes(options)
+            attributes = serializer.attributes
+            attributes.slice!(*options[:fields]) if options[:fields]
```

Comment:
This is actually the main change of this PR: instead of being handled at serializer level, this is now handled at adapter level which allows for different format per adapter.